Index: customer/apps/html/standard/page/wancfg.js
===================================================================
--- customer/apps/html/standard/page/wancfg.js	(revision 15771)
+++ customer/apps/html/standard/page/wancfg.js	(working copy)
@@ -324,6 +324,7 @@
 	var st_conn3g;
 	
 	if (G_WanConns[_i] && G_WanConns[_i][6] == '0') {
+       alert('IPv4, G_WanConns[_i][1] = ' + G_WanConns[_i][1]);
 		if(G_WanConns[_i] && G_WanConns[_i][1] == 'Connected'){
 			st_conn = 'Disconnect';
 		} else if(G_WanConns[_i] && G_WanConns[_i][1] != 'Connected'){
@@ -332,11 +333,11 @@
 			return false;
 		}
 	} else {
+       alert('IPv6, G_WanConns[_i][7] = ' + G_WanConns[_i][7]);
 		if(G_WanConns[_i] && G_WanConns[_i][7] == 'Disconnected')
 			st_conn = 'Connect';
 		else
 			st_conn = 'Disconnect';
-
 	}
 	
 	$H({
