Index: src/opensource/libnetfilter_conntrack/Makefile
===================================================================
--- src/opensource/libnetfilter_conntrack/Makefile	(revision 2007)
+++ src/opensource/libnetfilter_conntrack/Makefile	(working copy)
@@ -9,6 +9,8 @@
 		tar -xvf libnetfilter_conntrack-0.0.100.tar.bz2; \
 	fi
 	cd $(SRCDIR); \
+	rm -rf autom4te.cache/ aclocal.m4; \
+	autoreconf --install; \
 	find | xargs touch; \
 	export PKG_CONFIG_PATH=$(TMPDISTDIR)/lib/pkgconfig; \
 	LIBNFNETLINK_CFLAGS=-I$(TMPDISTDIR)/include \
