Index: tools/src/sqlzma/squashfs3.4/Makefile
===================================================================
--- tools/src/sqlzma/squashfs3.4/Makefile	(revision 2007)
+++ tools/src/sqlzma/squashfs3.4/Makefile	(working copy)
@@ -27,7 +27,7 @@
 sort.o: sort.c squashfs_fs.h global.h sort.h
 
 unsquashfs: unsquashfs.o
-	$(CC) $(LDFLAGS) unsquashfs.o -lz -lpthread -lm -lunlzma -o $@
+	$(CC) $(LDFLAGS) unsquashfs.o -lpthread -lm -lunlzma -lz -o $@
 
 unsquashfs.o: unsquashfs.c squashfs_fs.h read_fs.h global.h \
 	${Sqlzma}/sqlzma.h ${Sqlzma}/sqmagic.h ${LzmaC}/libunlzma.a
